### DISCUSSION TITLE
Error Encountered When Using High-Performance macOS Images in Azure Pipelines

### DISCUSSION THREAD

**Discussion by anderson-scherer** [actions/virtual-environments#10155] [General questions] [answered]
I would like to test higher performance images for Builds in Azure Pipelines.

```yml
pool:
  vmImage: 'macos-14-xlarge'
```

However, I am encountering the following error:

``` shell
##[warning]An image label with the label macos-14-xlarge does not exist.
##[error]The remote provider was unable to process the request.
```

What would be the procedure for using these higher performance images?

Thank you in advance for your assistance.

**Comment by pravinade** [comment, marked as answer]
You are currently using the GitHub Actions runner label "macos-14-xlarge" in your Azure Pipeline, hence it's throwing macos-14-xlarge does not exist. Please change it to "macOS-latest" or "macOS-14." So that It will be worked fine.For more information, please refer to the article: [Azure DevOps Hosted Agents](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops).

**Reply by kleverson1988** [reply]
O erro que você está recebendo acontece porque a imagem "macos-14-xlarge" não está disponível no Azure Pipelines, ou o nome está errado. Para resolver isso, siga estas etapas:

1. Verifique as imagens disponíveis: O Azure oferece imagens como:
macos-latest (mais recente)
macos-11 (Big Sur)
macos-12 (Monterey)
macos-13 (Ventura)

A imagem "macos-14-xlarge" não existe, então escolha uma dessas opções.

2. Escolha a imagem correta: Use algo como:
  vmImage: 'macos-latest'

Ou, se preferir uma versão específica, como o macOS Ventura:
  vmImage: 'macos-13'

4. Capacidade de máquina: Caso precise de mais desempenho, confira as opções de máquinas com mais recursos, mas geralmente, as imagens padrão já são suficientes.
5. Consulte a documentação: Se precisar de mais detalhes sobre as imagens e máquinas disponíveis, acesse a documentação do Azure Pipelines para ver a lista completa.

Resumindo, substitua "macos-14-xlarge" por uma imagem válida como "macos-latest" ou "macos-13" e ajuste a configuração conforme necessário
